shoes-core/lib/shoes/line.rb

```diff
@@ -19,6 +21,26 @@ def create_dimensions(point_a, point_b)
       style[:y2] = point_b.y
     end
 
+    # Check out http://math.stackexchange.com/questions/60070/checking-whether-a-point-lies-on-a-wide-line-segment
+    # for explanations how the algorithm works
+    def in_bounds?(x, y)
+      # c is (x, y)
+      left_most, right_most = point_a.x < point_b.x ? [point_a, point_b] : [point_b, point_a]
+      vec_left_c = Vector[left_most.x - x, left_most.y - y]
```

Comment:
hm. I'd probably rather have:

```
def +(other)
  Point.new(x - other.x, y - other.y)
end
```

Of course that'd require a `Point.new` each time around, your solution with to_a is quite elegant, though :)

I'm not too sure. Choose what you like best